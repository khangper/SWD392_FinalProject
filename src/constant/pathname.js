export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
