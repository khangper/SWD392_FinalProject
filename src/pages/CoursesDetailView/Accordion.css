.accordion-container {
  padding-top: 53px;
    font-family: Arial, sans-serif;
  }
  
.accordion-expand-holder span {
  margin-left: 15px;
  font-size: 15px;
}

  .accordion-header {
    
    display: flex;
    justify-content: space-between;
    padding: 10px;
    background-color: #f1f1f1;
    cursor: pointer;
    border-bottom: 1px solid #ccc;
  }
  
  .accordion-header:hover {
    background-color: #ddd;
  }
  
  .section-header-left {
    display: flex;
    align-items: center;
  }
  
  .section-title-wrapper {
    display: flex;
    align-items: center;
  }
  
  .crse_icon {
    margin-right: 10px;
  }
  
  .section-title-text {
    font-size: 16px;
    font-weight: bold;
  }
  
  .section-header-right {
    display: flex;
    align-items: center;
    font-size: 14px;
  }
  
  .num-items-in-section,
  .section-header-length {
    margin-left: 10px;
  }
  
  .accordion-content {
    background-color: #fff;
    border: 1px solid #ccc;
    border-top: none;
    padding: 10px;
  }
  
  .lecture-container {
    display: flex;
    justify-content: space-between;
    padding: 10px 0;
    border-bottom: 1px solid #eee;
  }
  
  .left-content {
    display: flex;
    align-items: center;
  }
  
  .icon_142 {
    margin-right: 10px;
  }
  
  .title {
    font-size: 14px;
  }
  
  .details {
    display: flex;
    align-items: center;
  }
  
  .preview-text {
    margin-right: 10px;
    color: #007bff;
    text-decoration: none;
  }
  
  .preview-text:hover {
    text-decoration: underline;
  }
  
  .content-summary {
    font-size: 12px;
    color: #888;
  }
  