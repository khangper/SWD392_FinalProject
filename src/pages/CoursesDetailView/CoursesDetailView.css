.courses_detail_view-box1 {
    flex: 0 0 auto;
    width: 33.33333333%;
    padding: 0 10px 0 10px;
}

.courses_detail_view-box2 {
    flex: 0 0 auto;
    width: 66.66666667%;
    padding: 0 10px 0 10px;

}

.courses_detail_view-row {
    display: flex;
    flex-wrap: wrap;
    gap: 20px;
    justify-content: space-between;
}

.courses_detail_view-row3 {
    display: flex;
    gap: 20px;
    justify-content: space-between;
}

.courses_detail_view-col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
}

.courses_detail_view-section {
    background-color: #333;
    padding: 20px;
    border-radius: 5px;
    color: #fff;
}

.courses_detail_view-row2 {
    display: flex;
}

.cdv-preview_video {
    background: #fff;
    padding: 10px;
    float: left;
    width: 100%;
    border-radius: 3px;
}

.cdv-preview_video .fcrse_img {
    width: 100%;
    float: left;
    position: relative;
}

.cdv-badge_seller {
    border-radius: 3px;
        font-size: 10px;
        position: absolute;
        padding: 2px 10px;
        background: #fa8305;
        color: #fff;
        font-family: "Roboto", sans-serif;
        font-weight: 600;
        right: 0;
        top: 1px;
        text-transform: uppercase;
        box-shadow: 0 0 1px 1px rgba(20, 23, 28, 0.1), 0 3px 1px 0 rgba(20, 23, 28, 0.1);
}

.cdv-label-preview-this-course {
    position: absolute;
    text-align: center;
    bottom: 0;
    color: #fff;
    font-family: 'Roboto', sans-serif;
    font-weight: 500;
    font-size: 14px;
    background: rgba(51, 51, 51, 0.9);
    left: 0;
    right: 0;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    padding: 10px 0;
}

.cdv-course-overlay {
    position: absolute;
    top: 10px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: flex-start;
}

.cdv-save {
    margin-top: 30px;
    color: #f3f3f3;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-align: left;
    float: left;
    margin-right: 30px;
}

.cdv-report {
    margin-top: 30px;
    color: #f3f3f3;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-align: left;
    float: left;
    margin-right: 30px;
}

.cdv-label1 h2 {
    margin-bottom: 15px;
}

.cdv-label2 {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Open Sans', sans-serif;
    color: #f7f7f7;
    text-align: left;
    line-height: 26px;
}

.cdv-rate_star {
    margin-top: 20px;
    color: #fff;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-align: left;
}

.cdv-language {
    display: flex;
    margin-top: 15px;
    color: #fff;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
}

.cdv-language-label1 {
    display: flex;
    margin-right: 20px;
}

.cdv-caption_tooltip {
    margin-left: 5px;
}

.cdv-caption-content {
    display: none;
    position: absolute;
    background-color: #fff;
    min-width: 200px;
    box-shadow: 0 2px 3px 0 rgba(34, 36, 38, .15);
    border: 1px solid rgba(34, 36, 38, .15);
    z-index: 1;
    /* right: 0; */
    border-radius: 5px;
}

.cdv-caption-content span {
    color: #686f7a;
    font-size: 12px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    padding: 5px 10px;
    text-decoration: none;
    display: block;
}

.cdv-caption_tooltip:hover .cdv-caption-content {
    display: block;
}

.cdv-add-buy {
    display: flex;
    margin-top: 27px;
}

.cdv-label-30day {
    margin-top: 10px;
    color: #fff;
    font-size: 12px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-align: left;
}

.courses_detail_view_tabs {
    /* margin-left: 320px; */
    display: inline;
    justify-content: flex-start;
    padding: 10px 20px;
    border-bottom: 1px solid #ddd;
}

.courses_detail_view-nav {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 10px 20px;
    border-bottom: 1px solid #ddd;
}

.courses_detail_view-div2 {
    float: left;
    width: 100%;
    padding: 20px 0 0;
    background: #fff;
    margin-bottom: 17px;
}

.cdv-user {
    float: left;
    width: 100%;
    margin-bottom: 20px;
}

.cdv-user_dt {
    float: left;
    
}

.cdv-live_user_dt {
    display: flex;
}

.cdv-user_cntnt{
    display: flex;
    flex-direction: column;
}

.cdv-contact_icon {
    display: flex;
    align-items: center;
    float: right;
}

.cdv-contact_icon img {
    width: 20px;
    margin-right: 10px;
}

.cdv-user_img {
    margin-right: 15px;
}

.cdv-user_img img {
    width: 50px;
}

.cdv-user-name {
    font-size: 16px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #333 !important;
    margin-top: 0;
    display: block;
    margin-bottom: 10px !important;
}

.cdv-subscribe-btn {
    border: 0;
    background: #ed2a26;
    padding: 0 20px;
    height: 30px;
    border-radius: 3px;
    color: #fff;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    font-size: 14px;
    cursor: pointer;
}

.cdv-contact_icon .cdv-report1 {
    display: flex;
    flex-direction: column;
    background: #fff;
    height: 60px;
    text-align: center;
    padding: 8px 15px;
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #686f7a;
    border: 1px solid #efefef;
    border-radius: 3px;

}

.cdv-report1 span {
    margin-top: 5px;
    display: flex;
    justify-content: space-between;
}



.tab-pane {
    padding: 20px;
}

._htg451 {
    margin: 0 auto;
    max-width: 800px;
}

._htg452 {
    margin-bottom: 35px;
}

h3 {
    font-size: 24px;
    margin-bottom: 20px;
    font-weight: 700;
}

ul {
    padding-left: 20px;
    list-style: none;
}

.courses_detail_view-box2 ul li {
    margin-bottom: 10px;
    line-height: 1.5;
}

ul._abc124 {
    padding-left: 0;
}

ul._abc124 li {
    position: relative;
    padding-left: 25px;
}

ul._abc124 li::before {
    content: "\2022";
    position: absolute;
    left: 0;
    color: #2c3e50;
    font-size: 20px;
    line-height: 1;
}

ul._abcd145 {
    display: flex;
    flex-direction: column;
    gap: 10px;
}

._5f7g11 {
    font-size: 16px;
    color: #333;
}

._5f7g15 {
    display: flex;
    align-items: center;
    gap: 10px;
}

._5f7g15 i {
    color: #27ae60;
    font-size: 18px;
}

.cdv-label-description p {
    font-size: 16px;
    line-height: 1.6;
    margin-bottom: 20px;
}

.col-lg-6 {
    width: 50%;
    float: left;
    box-sizing: border-box;
    padding-right: 20px;
}

.courses_detail_view-row::after {
    content: "";
    display: table;
    clear: both;
}

@media screen and (max-width: 768px) {
    .col-lg-6 {
        width: 100%;
        padding-right: 0;
    }
}

.box-div-cdv {
    background-color: #f9f4f4;
}

.cdv-label-requirement {
    background-color: #f9f4f4;
}

.cdv-label-description {
    background-color: #f9f4f4;
}

.cdv-label-who {
    background-color: #f9f4f4;
}

.accordion-toggle {
    height: 0px;
    font-size: 18px;
    opacity: 0;
    transition:  
    opacity 0.3s ease-in-out,
    height 0.3s 0.3s ease-in-out;
    color: #333;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
}

.animated {
    opacity: 1;
    transition:  
    height 0.3s ease-in-out,
    opacity 0.3s 0.3s ease-in-out;
}

.accordion-toggle p {
    margin: 0;
    padding: 22px 15px;
    pointer-events: none;
    line-height: 1.3;
}

.accordion-visible {
    background: #7177ca;
    width: 100%;
    color: #fff;
    cursor: pointer;
    border: 1px solid #474d9b;;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 7px 15px;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
}

.lecture-container {
    display: flex;
    padding: 12px 30px 12px 20px;
    background-color: #fff;
    border-bottom: 1px solid #efefef;
    border-top: none;
    color: #686f7a;
    letter-spacing: .3px;
    line-height: 1.33;
    font-weight: 400;
    margin-left: 0 !important;
}

.cdv-accordion-label {
    display: flex;
    float: right;
}



.cdv-crse_content h3 {
    font-size: 20px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    text-align: left;
    margin-bottom: 15px;
    color: #333;
    float: left;
}

.cdv-label-20 {
    float: left;
    margin-top: 5px;
    border: 1px solid #dedfe0;
    height: 50px;
    background: #e8e9eb;
    width: 100%;
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #333;
    text-align: center;
    padding: 15px 0;
}

.student_reviews {
    float: left;
    width: 100%;
    margin-top: 30px;
    margin-bottom: 20px;
}

.reviews-container {
    display: flex;
    flex-wrap: wrap;
  }
  
  .courses_detail_view-row {
    display: flex;
    justify-content: space-between;
  }
  
  .cdv-col-lg-5, .cdv-col-lg-7 {
    padding: 20px;
  }

  .cdv-col-lg-5 {
    width: 28%;
  }
  
  .cdv-col-lg-7 {
    width: 70%;
  }

  .reviews_left, .review_right {
    background-color: #f9f9f9;
    padding: 20px;
    border-radius: 8px;
  }
  
  .total_rating {
    display: flex;
    align-items: center;
    margin-bottom: 20px;
  }
  
  ._rate001 {
    font-size: 2em;
    margin-right: 10px;
  }
  
  .rating-box {
    display: flex;
    align-items: center;
  }
  
  .cdv-rate-star {
    width: 20px;
    margin-right: 2px;
  }
  
  ._rate002 {
    margin-left: 10px;
    font-size: 1em;
  }
  
  ._rate003 {
    margin-top: 10px;
  }
  
  ._rate004 {
    display: flex;
    align-items: center;
    margin-bottom: 10px;
  }
  
  .progress {
    width: 100px;
    height: 10px;
    background-color: #e0e0e0;
    margin-right: 10px;
  }
  
  .progress-bar {
    height: 10px;
    background-color: red;
  }
  
  .review_right_heading {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  
  .review_search {
    display: flex;
    align-items: center;
  }
  
  .rv_srch {
    padding: 5px;
    border: 1px solid #ccc;
    border-radius: 4px;
    margin-right: 5px;
  }
  
  .rvsrch_btn {
    
    color: white;
    border: none;
    padding: 5px 10px;
    border-radius: 4px;
    cursor: pointer;
  }
  .rvsrch_btn img{
    width: 20px;
  }
  .review_all120 {
    margin-top: 20px;
  }
  
  .review_item {
    padding: 15px 0;
    border-bottom: 1px solid #ddd;
  }
  
  .review_usr_dt {
    display: flex;
    align-items: center;
    margin-bottom: 10px;
  }
  
  .cdv-cmnt_text_message {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    margin-top: 17px;
    line-height: 26px;
    color: #686f7a;
    text-align: left;
}

  .cdv-check-helpful {
    display: flex;
    align-items: center;
    padding-top: 8px;
    /* justify-content: space-between; */
  }
  
  .cdv-check-helpful span {
    margin-right: 18px;
    font-size: 14px;
  }

  .radio--group-inline-container {
    display: flex;
  }
  
  .radio-item {
    font-size: 14px;
    margin-right: 18px;
  }
  
  .radio-item input[type="radio"]:checked  {
    accent-color: red;
  }

  .radio-label {
    margin-left: 5px;
  }
  
  .report145 {
    /* color: red; */
    cursor: pointer;
  }
  
  .more_reviews {
    display: block;
    text-align: center;
    /* color: red; */
    margin-top: 20px;
  }

  .starIcon {
    width: 15px;
    height: 15px;
    top: 1px;
    position: relative;
    margin-right: 5px;
  }