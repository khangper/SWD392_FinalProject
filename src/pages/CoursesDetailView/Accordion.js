import React, { useState } from 'react';
import './Accordion.css';

const Accordion = () => {
    const [activeIndex, setActiveIndex] = useState(null);

    const toggleAccordion = (index) => {
        setActiveIndex(activeIndex === index ? null : index);
    };

    return (
        <div className="accordion-container">
            {sections.map((section, index) => (
                <div key={index}>
                    <div className="accordion-header" onClick={() => toggleAccordion(index)}>
                        <div className="section-header-left">
                            <span className="section-title-wrapper">
                                <i className="uil uil-presentation-play crse_icon"></i>
                                <span className="section-title-text">{section.title}</span>
                            </span>
                        </div>
                        <div className="section-header-right">
                            <span className="num-items-in-section">{section.lectures.length} lectures</span>
                            <span className="section-header-length">{section.length}</span>
                        </div>
                    </div>
                    {activeIndex === index && (
                        <div className="accordion-content">
                            {section.lectures.map((lecture, idx) => (
                                <div className="lecture-container" key={idx}>
                                    <div className="left-content">
                                        <i className={`uil ${lecture.icon} icon_142`}></i>
                                        <div className="top">
                                            <div className="title">{lecture.title}</div>
                                        </div>
                                    </div>
                                    <div className="details">
                                        {lecture.preview && <a href="#" className="preview-text">Preview</a>}
                                        <span className="content-summary">{lecture.summary}</span>
                                    </div>
                                </div>
                            ))}
                        </div>
                    )}
                </div>
            ))}
        </div>
    );
};

const sections = [
    {
        title: 'Introduction to this Course',
        length: '22:08',
        lectures: [
            { title: 'A Note On Asking For Help', summary: '00:12', icon: 'uil-file' },
            { title: 'Introducing Our TA', summary: '01:42', icon: 'uil-file' },
            { title: 'Join the Online Community', summary: '00:51', icon: 'uil-file' },
            { title: 'Why This Course?', summary: '07:48', icon: 'uil-play-circle', preview: true },
            { title: 'Syllabus Download', summary: '2 pages', icon: 'uil-file-download-alt', preview: true },
            { title: 'Syllabus Walkthrough', summary: '01:42', icon: 'uil-play-circle', preview: true },
            { title: 'Lecture Slides', summary: '00:11', icon: 'uil-file' },
        ],
    },
    {
        title: 'Introduction to Front End Development',
        length: '27:26',
        lectures: [
            { title: 'Unit Objectives', summary: '01.40', icon: 'uil-play-circle', preview: true },
            { title: 'Note about Setting Up Front-End Developer Environment', summary: '00:30', icon: 'uil-file' },
            { title: 'Setting Up Front-End Developer Environment', summary: '03:11', icon: 'uil-play-circle' },
            { title: 'Note about Introduction to the Web', summary: '00:11', icon: 'uil-file' },
            { title: 'Introduction to the Web', summary: '10.08', icon: 'uil-play-circle' },
            { title: 'The Front End Holy Trinity', summary: '11:46', icon: 'uil-play-circle', preview: true },
        ],
    },
    {
        title: 'Introduction to HTML',
        length: '58:55',
        lectures: [
            { title: 'Unit Objectives', summary: '01:38', icon: 'uil-play-circle' },
            { title: 'HTML Basics and Setting Up Front-End Developer Environment', summary: '05:53', icon: 'uil-play-circle' },
            { title: 'Note about Introduction to MDN', summary: '00:19', icon: 'uil-file' },
            { title: 'Introduction to MDN', summary: '01:51', icon: 'uil-play-circle' },
            { title: 'HTML Boilerplate and Comments', summary: '09:42', icon: 'uil-play-circle' },
            { title: 'Basic Tags', summary: '08:23', icon: 'uil-play-circle' },
            { title: 'HTML Lists', summary: '04:32', icon: 'uil-play-circle' },
            { title: 'HTML Lists Assignment', summary: '01:23', icon: 'uil-play-circle' },
            { title: 'HTML Lists Assignment: SOLUTION', summary: '05:59', icon: 'uil-play-circle' },
            { title: 'Divs and Spans', summary: '05:23', icon: 'uil-play-circle' },
            { title: 'HTML Attributes', summary: '09:00', icon: 'uil-play-circle' },
            { title: 'Recreate Webpage Assignment', summary: '01:00', icon: 'uil-play-circle' },
            { title: 'Recreate Webpage Assignment: SOLUTION', summary: '03:56', icon: 'uil-play-circle' }
        ],
    },
    {
        "title": "Intermediate HTML",
        "length": "01:12:29",
        "lectures": [
            { "title": "Unit Objectives", "summary": "01:19", "icon": "uil-play-circle" },
            { "title": "HTML Tables", "summary": "06:03", "icon": "uil-play-circle" },
            { "title": "Tables Exercise", "summary": "01:18", "icon": "uil-file" },
            { "title": "Introduction to Forms", "summary": "06:19", "icon": "uil-play-circle" },
            { "title": "Playing with Inputs", "summary": "03:04", "icon": "uil-play-circle" },
            { "title": "The Form Tag", "summary": "06:36", "icon": "uil-play-circle" },
            { "title": "Labels", "summary": "03:32", "icon": "uil-play-circle" },
            { "title": "Form Validations", "summary": "03:43", "icon": "uil-play-circle" },
            { "title": "Dropdowns and Radio Buttons", "summary": "11:19", "icon": "uil-play-circle" },
            { "title": "Form Exercise", "summary": "07:23", "icon": "uil-play-circle" }
        ]
    },
    {
        "title": "Introduction to CSS",
        "length": "01:40:15",
        "lectures": [
            { "title": "Unit Objectives", "summary": "02:28", "icon": "uil-play-circle" },
            { "title": "CSS Basics", "summary": "05:25", "icon": "uil-play-circle" },
            { "title": "Our First Stylesheet", "summary": "10:18", "icon": "uil-file" },
            { "title": "Note about CSS Colors and Background and Border (next 2 lectures)", "summary": "00:20", "icon": "uil-file" },
            { "title": "CSS Colors", "summary": "14:35", "icon": "uil-play-circle" },
            { "title": "Background and Border", "summary": "06:58", "icon": "uil-play-circle" },
            { "title": "Note about Selectors Basics Todo List", "summary": "00:32", "icon": "uil-file" },
            { "title": "Selectors Basics Todo List", "summary": "10:43", "icon": "uil-play-circle" },
            { "title": "Introduction to Chrome Inspector", "summary": "05:43", "icon": "uil-play-circle" },
            { "title": "More Advanced Selectors", "summary": "09:23", "icon": "uil-play-circle" },
            { "title": "Note about Specificity and the Cascade", "summary": "00:23", "icon": "uil-file" },
            { "title": "Specificity and the Cascade", "summary": "10:38", "icon": "uil-play-circle" },
            { "title": "Note about Selectors Practice Exercise", "summary": "00:22", "icon": "uil-file" },
            { "title": "Selectors Practice Exercise", "summary": "04:28", "icon": "uil-play-circle" },
            { "title": "Selectors Practice Exercise: SOLUTION", "summary": "16:48", "icon": "uil-play-circle" }
        ]
    },
    {
        "title": "Intermediate CSS",
        "length": "01:26:40",
        "lectures": [
            { "title": "Unit Objectives", "summary": "01:40", "icon": "uil-play-circle" },
            { "title": "Text and Fonts", "summary": "09:55", "icon": "uil-play-circle" },
            { "title": "More Text and Fonts", "summary": "04:42", "icon": "uil-file" },
            { "title": "Note about Google Fonts", "summary": "00:22", "icon": "uil-file" },
            { "title": "Using Google Fonts", "summary": "04:35", "icon": "uil-play-circle" },
            { "title": "Introduction to the Box Model", "summary": "11:58", "icon": "uil-play-circle" },
            { "title": "Creating a Tic Tac Toe Board", "summary": "01:41", "icon": "uil-play-circle" },
            { "title": "Creating a Tic Tac Toe Board: SOLUTION", "summary": "07:43", "icon": "uil-play-circle" },
            { "title": "Note about Image Gallery Code Along Pt. 1", "summary": "00:20", "icon": "uil-file" },
            { "title": "Image Gallery Code Along Pt. 1", "summary": "08:20", "icon": "uil-play-circle" },
            { "title": "Note about changes to Google Fonts", "summary": "00:40", "icon": "uil-file" },
            { "title": "Image Gallery Code Along Pt. 2", "summary": "05:46", "icon": "uil-play-circle" },
            { "title": "CSS Blog From Scratch Exercise Intro", "summary": "03:23", "icon": "uil-play-circle" },
            { "title": "CSS Blog From Scratch Exercise SOLUTION Pt. 1", "summary": "08:38", "icon": "uil-play-circle" },
            { "title": "CSS Blog From Scratch Exercise SOLUTION Pt. 2", "summary": "08:12", "icon": "uil-file" },
            { "title": "CSS Blog From Scratch Exercise SOLUTION Pt. 3", "summary": "07:28", "icon": "uil-play-circle" }
        ]
    },
    {
        "title": "Bootstrap",
        "length": "01:59:54",
        "lectures": [
            { "title": "Unit Objectives", "summary": "02:28", "icon": "uil-play-circle" },
            { "title": "Note about Bootstrap versions", "summary": "00:44", "icon": "uil-file" },
            { "title": "What is Bootstrap?", "summary": "08:02", "icon": "uil-play-circle" },
            { "title": "Adding Bootstrap to a Project", "summary": "09:15", "icon": "uil-play-circle" },
            { "title": "Forms and Inputs", "summary": "14:00", "icon": "uil-play-circle" },
            { "title": "Nav Bars", "summary": "15:44", "icon": "uil-play-circle" },
            { "title": "Note about The Grid System", "summary": "00:11", "icon": "uil-file" },
            { "title": "The Grid System", "summary": "08:43", "icon": "uil-play-circle" },
            { "title": "Grid System Pt. 2", "summary": "11:43", "icon": "uil-play-circle" },
            { "title": "Note about Bootstrap Image Gallery Pt. 1", "summary": "00:55", "icon": "uil-file" },
            { "title": "Bootstrap Image Gallery Pt. 1", "summary": "14:20", "icon": "uil-play-circle" },
            { "title": "Note about Font Awesome", "summary": "00:16", "icon": "uil-file" },
            { "title": "Bootstrap Image Gallery Pt. 2", "summary": "09:46", "icon": "uil-play-circle" },
            { "title": "Creating a Startup Landing Page Code Along", "summary": "14:23", "icon": "uil-play-circle" },
            { "title": "Creating a Startup Landing Page Code Along Pt. 2", "summary": "10:30", "icon": "uil-play-circle" },
            { "title": "Note about making bootstrap responsive on mobile devices", "summary": "00:14", "icon": "uil-file" }
        ]
    },
    {
        "title": "Bootstrap 4!",
        "length": "01:16:28",
        "lectures": [
            { "title": "A History of Bootstrap 4", "summary": "04:40", "icon": "uil-play-circle" },
            { "title": "The Bootstrap 4 Documentation", "summary": "02:24", "icon": "uil-play-circle" },
            { "title": "Changes from Bootstrap 3 to 4", "summary": "07:52", "icon": "uil-play-circle" },
            { "title": "Bootstrap 4 Code/Solutions Download", "summary": "00:41", "icon": "uil-file" },
            { "title": "Getting Started With Bootstrap 4", "summary": "05:15", "icon": "uil-play-circle" },
            { "title": "Bootstrap 4 Colors and Backgrounds", "summary": "05:59", "icon": "uil-play-circle" },
            { "title": "Typography", "summary": "11:12", "icon": "uil-play-circle" },
            { "title": "New Fancy Spacing Utilities", "summary": "12:28", "icon": "uil-play-circle" },
            { "title": "Responsive Breakpoints", "summary": "10:55", "icon": "uil-play-circle" },
            { "title": "Bootstrap4 Navbars", "summary": "08:20", "icon": "uil-play-circle" },
            { "title": "The New Display Utility", "summary": "07:26", "icon": "uil-play-circle" },
            { "title": "Bootstrap 4: Flexbox and Layout", "summary": "10:14", "icon": "uil-play-circle" },
            { "title": "Flexbox Utilities Part 2", "summary": "07:23", "icon": "uil-play-circle" },
            { "title": "Navs and Flexbox", "summary": "09:56", "icon": "uil-play-circle" },
            { "title": "The Bootstrap 4 Grid", "summary": "08:56", "icon": "uil-play-circle" },
            { "title": "PATTERN PROJECT Part 1", "summary": "12:06", "icon": "uil-play-circle" },
            { "title": "PATTERN PROJECT Part 2", "summary": "11:30", "icon": "uil-play-circle" },
            { "title": "The Grid + Flexbox", "summary": "11:44", "icon": "uil-play-circle" },
            { "title": "Museum of Candy Project Part 1", "summary": "05:36", "icon": "uil-play-circle" },
            { "title": "Note about Museum of Candy Project Part 2", "summary": "00:12", "icon": "uil-file" }
        ]
    },
    {
        "title": "Introduction to JavaScript",
        "length": "56:21",
        "lectures": [
            { "title": "Unit Objectives", "summary": "04:41", "icon": "uil-play-circle" },
            { "title": "The JavaScript Console", "summary": "02:22", "icon": "uil-play-circle" },
            { "title": "Primitives", "summary": "13:22", "icon": "uil-play-circle" },
            { "title": "Primitives Exercises", "summary": "03:21", "icon": "uil-file" },
            { "title": "Variables", "summary": "06:15", "icon": "uil-play-circle" },
            { "title": "Null and Undefined", "summary": "02:33", "icon": "uil-play-circle" },
            { "title": "Useful Built-In Methods", "summary": "05:12", "icon": "uil-play-circle" },
            { "title": "Writing JavaScript in a Separate File", "summary": "04:28", "icon": "uil-play-circle" },
            { "title": "JS Stalker Exercise", "summary": "01:55", "icon": "uil-play-circle" },
            { "title": "JS Stalker Exercise: SOLUTION", "summary": "04:45", "icon": "uil-play-circle" },
            { "title": "Age Calculator Exercise", "summary": "01:10", "icon": "uil-play-circle" },
            { "title": "Age Calculator Exercise: SOLUTION", "summary": "04:01", "icon": "uil-play-circle" }
        ]
    }





];

export default Accordion;
