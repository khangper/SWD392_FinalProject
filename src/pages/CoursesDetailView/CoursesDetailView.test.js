import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import CoursesDetailView from "./CoursesDetailView";
import { screen } from "@testing-library/react";

describe("CoursesDetailView", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the CoursesDetailView component", async () => {
        const { container } = renderWithProviders(<CoursesDetailView />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });
})