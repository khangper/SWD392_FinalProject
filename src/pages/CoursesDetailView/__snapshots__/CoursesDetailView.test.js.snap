// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword renders the ReportHistory component 1`] = `
<div>
  <div>
    Course not found
  </div>
</div>
`;
